lib/OpenQA/WebAPI/Plugin/AMQP.pm

```diff
@@ -70,6 +71,179 @@ sub log_event {
+            # so we construct it out of BUILD and the scenario keys.
+            # 'name' is supposed to be a 'human readable name', well,
+            # this is human readable, so we'll just use it twice
+            id   => join('.', "openqa", $job->BUILD, $job->TEST, $job->MACHINE, $job->FLAVOR, $job->ARCH),
+            name => join('.', "openqa", $job->BUILD, $job->TEST, $job->MACHINE, $job->FLAVOR, $job->ARCH),
+        },
+        test => {
+            # openQA tests are pretty much always validation
+            category => 'validation',
+            # test identifier: test name plus scenario keys
+            type      => join(' ', $job->TEST, $job->MACHINE, $job->FLAVOR, $job->ARCH),
+            namespace => $test_namespace,
+        },
+        system => {
+            # it's interesting whether we should record info on the
+            # *worker host itself* or the *SUT* (the VM run on top of
+            # the worker host environment) here...on the whole I think
+            # SUT is more in line with expectations, so let's do that
+            os => "fedora-${sysrelease}",
+            # openqa provisions itself...we *could* I guess set this
+            # to 'createhdds' if we booted a disk image, but ehhhh
+            provider     => 'openqa',
+            architecture => $job->ARCH,
+            variant      => $job->settings_hash->{SUBVARIANT},
+        },
+        generated_at => $generated_at,
+        version      => "0.2.1",
+    );
+
+    # add keys that don't exist in all cases to the message
+    if ($stdevent eq 'complete') {
+        $msg_data{test}{result} = $job->result;
+        $msg_data{test}{result} = 'info' if $job->result eq 'softfailed';
+    }
+    elsif ($stdevent eq 'error') {
+        $msg_data{error} = {};
+        $msg_data{error}{reason} = $job->result;
+    }
+    elsif ($stdevent eq 'queued') {
+        # this is a hint to consumers that the job probably went away
+        # if they don't get a 'complete' or 'error' in 4 hours
+        # FIXME: we should set this as 2 hours on 'running', but we
+        # can't emit running ATM...
+        $msg_data{test}{lifetime} = 240;
+    }
+    $msg_data{run}{clone_of} = $clone_of if ($clone_of);
+
+    $msg_data{artifact}{release} = $artifact_release if ($artifact_release);
+
+    $msg_data{artifact}{compose_type} = $compose_type if ($compose_type);
+
+    $msg_data{artifact}{iso} = $job->settings_hash->{ISO} if ($job->settings_hash->{ISO});
+    # FIXME: hdd_2?
+    $msg_data{artifact}{hdd_1} = $job->settings_hash->{HDD_1} if ($job->settings_hash->{HDD_1});
+
+    # convert data to JSON, with reliable key ordering (helps the tests)
+    my $msg_json = Cpanel::JSON::XS->new->canonical(1)->allow_blessed(1)->encode(\%msg_data);
+
+    # create the topic
+    my $topic = "ci.$artifact.test.$stdevent";
+
+    # finally, send the message
+    OpenQA::Utils::log_debug("Sending CI Messages AMQP message for $event");
```

Comment:
Why is `log_debug` not just imported?

Reply:
no particular reason...in fact other places in the file just use `log_debug` so I guess I could change it. I think this may be copied from the fedmsg plugin.